consensus/scripts/nbconverted/build-consensus-signatures.py

```diff
@@ -0,0 +1,247 @@
+
+# In[6]:
+
+
+# Load Data
+all_profiles_df = []
+for plate_dir in plate_dirs:
+    plate = plate_dir.name
+    plate_file = plate_dir / f"{plate}{file_base}"
+    plate_df = pd.read_csv(plate_file)
+    all_profiles_df.append(plate_df)
+
+# Concatenate profiles
+all_profiles_df = pd.concat(all_profiles_df, axis="rows")
+
+# Determine every CellProfiler feature measured
+cp_features = infer_cp_features(all_profiles_df)
+
+print(all_profiles_df.shape)
+all_profiles_df.head()
+
+
+# ## Recode Dose
+
+# In[7]:
+
+
+# Recode dose
+all_profiles_df = all_profiles_df.assign(
+    Metadata_dose_recode=(
+        all_profiles_df.Metadata_mmoles_per_liter.apply(
+            lambda x: recode_dose(x, primary_dose_mapping, return_level=True)
+        )
+    )
+)
+
+# Make sure DMSO profiles recieve a zero dose level
+all_profiles_df.loc[
+    all_profiles_df.Metadata_broad_sample == "DMSO", "Metadata_dose_recode"
+] = 0
+
+
+# In[8]:
+
+
+all_profiles_df.Metadata_dose_recode.value_counts()
+
+
+# In[9]:
+
+
+all_profiles_df.plot(
+    x="Metadata_mmoles_per_liter", y="Metadata_dose_recode", kind="scatter"
+)
+
+
+# ## Create Consensus Profiles
+# 
+# ### a) Generate consensus profiles for DMSO by well
+
+# In[10]:
+
+
+dmso_replicate_cols = [
+    "Metadata_Plate_Map_Name",
+    "Metadata_broad_sample",
+    "Metadata_pert_well",
+    "Metadata_mmoles_per_liter",
+    "Metadata_dose_recode",
+]
+
+# Isolate DMSO profiles
+dmso_profile_df = all_profiles_df.query("Metadata_broad_sample == 'DMSO'").reset_index(
+    drop=True
+)
+
+# How many DMSO profiles per well?
+print(dmso_profile_df.shape)
+
+pd.crosstab(dmso_profile_df.Metadata_pert_well, dmso_profile_df.Metadata_Plate_Map_Name)
+
+
+# In[11]:
+
+
+# Form dmso consensus profiles with median and modz
+dmso_profiles = {}
+for operation in operations:
+    dmso_profiles[operation] = consensus_apply(
+        dmso_profile_df,
+        operation=operation,
+        cp_features=cp_features,
+        replicate_cols=dmso_replicate_cols,
+    )
+
+
+# ### b) Generate consensus profiles per compound-dose pair
+
+# In[12]:
+
+
+compound_replicate_cols = [
+    "Metadata_Plate_Map_Name",
+    "Metadata_broad_sample",
+    "Metadata_mmoles_per_liter",
+    "Metadata_dose_recode",
+]
+
+compound_profile_df = all_profiles_df.query(
+    "Metadata_broad_sample != 'DMSO'"
+).reset_index(drop=True)
+
+
+# In[13]:
+
+
+# Form compound consensus profiles with median and modz
+compound_profiles = {}
+for operation in operations:
+    compound_profiles[operation] = consensus_apply(
+        compound_profile_df,
+        operation=operation,
+        cp_features=cp_features,
+        replicate_cols=compound_replicate_cols,
+    )
+
+    compound_profiles[operation] = compound_profiles[operation].assign(
+        Metadata_pert_well="collapsed"
+    )
+
+
+# ## Merge and Output Consensus Signatures
+
+# In[14]:
+
+
+for operation in operations:
+    consensus_file = f"{operation}_consensus_signatures_level5.csv.gz"
```

Comment:
Do you prefer this to the suggested name here https://github.com/broadinstitute/lincs-cell-painting/pull/40#issuecomment-630480994 (use batch name, similar to the way you'd use plate name)

Reply:
oh yes, overlooked that, my fault - will update in next commit!